python/paddle/fluid/transpiler/distribute_transpiler.py

```diff
@@ -1245,8 +1245,8 @@ def _get_optimize_pass(self):
                 # ops, we may get the output of clip op. Use syntax "@GRAD"
                 # and op_role_var to get the pair.
                 for input_name in op.input_arg_names:
-                    if input_name.find("@GRAD") != -1 and \
-                        op.attrs[RPC_OP_ROLE_ATTR_NAME]:
+                    if input_name.find("@GRAD") != -1 and op.attrs[
+                            RPC_OP_ROLE_ATTR_NAME] and "LearningRate" in op.input_names:
```

Comment:
This may not be the correct fix I suppose, this line was not trying to find the optimizer ops like "sgd", but to find any op that have the "optimizer" role and to get the original gradient variable name out of it.

Reply:
I may need to test for your case.

Reply:
I will test it with SGD.

Reply:
SGD is OK too.